### PR TITLE
Add GA4

### PR DESCRIPTION
**Pull request by qtamu**
# Description
Added Google Tag Manager scripts using an injection script in docker-entrypoint.sh to replace a placeholder in the index.html file. It will only run the script when nodenv is set to production.

Fixes # (issue) 
KACE ticket to add the Google Analytics tags .

## Type of change
- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Created a test google tag manager account to use the container id. I used the container id in my injection script then built the image when the nodeenv is set to prod and dev to see if  the script is ran properly and my conditional statement works.

**Test Configuration**:
* Toolchain:
* SDK:

# Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes
- [x ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ ARG USER_ID=3001
 ARG USER_NAME=sage
 ARG SOURCE_DIR=/$USER_NAME/source
 ARG NPM_REGISTRY=upstream
-ARG NODE_ENV=development
```
Comment (rmathew1011) [inline Dockerfile:6]: I would get this confirmed with Micah. So far, this has been the default and its also environment specific. The modification is prominent during the deployment phase ( and thats environment specific)

File: Dockerfile [hunk 2]
```diff
@@ -65,14 +65,16 @@ RUN echo $NPM_REGISTRY && \
...
 
-RUN ga4=$(cat ./build/Ga4.txt) && \
-    gtm=$(cat ./build/Gtm.txt) && \
-    ga4_one_line=$(echo "$ga4" | tr -d '\n') && \
-    gtm_one_line=$(echo "$gtm" | tr -d '\n') && \
-    ga4_escaped=$(echo "$ga4_one_line" | sed -e 's/[\/&]/\\&/g') && \
-    gtm_escaped=$(echo "$gtm_one_line" | sed -e 's/[\/&]/\\&/g') && \
-    sed -i "s#<!--google Analytics Tag -->#${ga4_escaped}#g" $SOURCE_DIR/src/main/resources/templates/index.html && \
-    sed -i "s#<!-- Google Tag Manager (noscript) -->#${gtm_escaped}#g" $SOURCE_DIR/src/main/resources/templates/index.html
+RUN if [ "$NODE_ENV" = "production" ]; then \
```
Comment (rmathew1011) [inline Dockerfile:68]: This condition is set only for the Prod env - would there be conditions for other env's as well

**Review by rmathew1011 — CHANGES_REQUESTED** [review]
`example.env` serves as a template - please confirm the same prior to deleting this file.

File: Dockerfile (hunk 2 above)
Comment (qtamu) [reply Dockerfile:68]: No, Just when the ENV is set to production.

File: Dockerfile (hunk 1 above)
Comment (qtamu) [reply Dockerfile]: I spoke with Micha and made the corrected changes.

**Review by rmathew1011 — APPROVED** [review]

**Pull request closed without merging**